update-grackle.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+if [[ -z "$IN_NIX_SHELL" ]]; then 
+    # Not in nix shell, use the default go installs
+    GRACKLE="go run github.com/mjschwenne/grackle/cmd/grackle@latest"
+else 
+    # In a nix shell, expect grackle to be on the PATH and use that
+    GRACKLE="grackle"
+fi
+
+# Have I mentioned that I dislike bash? Just look at this arcane
+# and archaic syntax
```

Comment:
Would this be nicer as a python script (especially if this gets more complex in the future)? E.g. something like Perennial's `etc/update-goose.py`.

Reply:
I also would prefer to migrate this to python now rather than later. It's already at my personal limit for bash script complexity.

Reply:
I would also prefer that, although it was @tchajed who suggested bash in the first place. I will work on this shortly.

Reply:
Sorry for flip flopping on that, I thought this would be simpler than it turned out.